int32 my_phase
bool isFinished
---
bool isMyTurn